pandera/dtypes_.py

```diff
@@ -0,0 +1,286 @@
+        if self.__class__ is _Number:
+            return isinstance(datatype, (Int, Float, Complex))
+        return super().check(datatype)
+
+
+@immutable
+class _PhysicalNumber(_Number):
+    bit_width: int = None
+    _base_name: str = field(default=None, init=False, repr=False)
+
+    def __eq__(self, obj: object) -> bool:
+        if isinstance(obj, type(self)):
+            return obj.bit_width == self.bit_width
+        return super().__eq__(obj)
+
+    def __str__(self) -> str:
+        return f"{self._base_name}{self.bit_width}"
+
+
+################################################################################
+## signed integer
+################################################################################
+
+
+@immutable(eq=False)
+class Int(_PhysicalNumber):
+    _base_name = "int"
+    continuous = False
+    exact = True
+    bit_width = 64
+    signed: bool = field(default=True, init=False)
+
+
+@immutable
+class Int64(Int, _PhysicalNumber):
+    bit_width = 64
+
+
+@immutable
+class Int32(Int64):
+    bit_width = 32
+
+
+@immutable
+class Int16(Int32):
+    bit_width = 16
+
+
+@immutable
+class Int8(Int16):
+    bit_width = 8
+
+
+################################################################################
+## unsigned integer
+################################################################################
+
+
+@immutable
+class UInt(Int):
+    _base_name = "uint"
+    signed: bool = field(default=False, init=False)
+
+
+@immutable
+class UInt64(UInt):
+    bit_width = 64
+
+
+@immutable
+class UInt32(UInt64):
+    bit_width = 32
+
+
+@immutable
+class UInt16(UInt32):
+    bit_width = 16
+
+
+@immutable
+class UInt8(UInt16):
+    bit_width = 8
+
+
+################################################################################
+## float
+################################################################################
+
+
+@immutable(eq=False)
+class Float(_PhysicalNumber):
+    _base_name = "float"
+    continuous = True
+    exact = False
+    bit_width = 64
+
+
+@immutable
+class Float128(Float):
+    bit_width = 128
+
+
+@immutable
+class Float64(Float128):
+    bit_width = 64
+
+
+@immutable
+class Float32(Float64):
+    bit_width = 32
+
+
+@immutable
+class Float16(Float32):
+    bit_width = 16
+
+
+################################################################################
+## complex
+################################################################################
+
+
+@immutable(eq=False)
+class Complex(_PhysicalNumber):
+    _base_name = "complex"
+    bit_width = 128
+
+
+@immutable
+class Complex256(Complex):
+    bit_width = 256
+
+
+@immutable
+class Complex128(Complex):
+    bit_width = 128
+
+
+@immutable
+class Complex64(Complex128):
+    bit_width = 64
+
+
+################################################################################
+# nominal
+################################################################################
+
+
+@dataclass(frozen=True)
```

Comment:
why not use `immutable` here?

Reply:
Initially I didn't have the `immutable` helper :sweat_smile: